Source/BoundaryConditions/PML.cpp

```diff
@@ -495,12 +495,25 @@ PML::PML (const BoxArray& grid_ba, const DistributionMapping& /*grid_dm*/,
         ngf = ngFFT;
     }
 
-    pml_E_fp[0] = std::make_unique<MultiFab>(amrex::convert( ba,
-        WarpX::GetInstance().getEfield_fp(0,0).ixType().toIntVect() ), dm, 3, nge );
-    pml_E_fp[1] = std::make_unique<MultiFab>(amrex::convert( ba,
-        WarpX::GetInstance().getEfield_fp(0,1).ixType().toIntVect() ), dm, 3, nge );
-    pml_E_fp[2] = std::make_unique<MultiFab>(amrex::convert( ba,
-        WarpX::GetInstance().getEfield_fp(0,2).ixType().toIntVect() ), dm, 3, nge );
+    if (do_dive_cleaning)
+    {
+        pml_E_fp[0] = std::make_unique<MultiFab>(amrex::convert( ba,
+            WarpX::GetInstance().getEfield_fp(0,0).ixType().toIntVect() ), dm, 3, nge );
+        pml_E_fp[1] = std::make_unique<MultiFab>(amrex::convert( ba,
+            WarpX::GetInstance().getEfield_fp(0,1).ixType().toIntVect() ), dm, 3, nge );
+        pml_E_fp[2] = std::make_unique<MultiFab>(amrex::convert( ba,
+            WarpX::GetInstance().getEfield_fp(0,2).ixType().toIntVect() ), dm, 3, nge );
+    }
+    else // diagonal field components (xx,yy,zz) not needed when do_dive_cleaning = false
+    {
+        pml_E_fp[0] = std::make_unique<MultiFab>(amrex::convert( ba,
+            WarpX::GetInstance().getEfield_fp(0,0).ixType().toIntVect() ), dm, 2, nge );
+        pml_E_fp[1] = std::make_unique<MultiFab>(amrex::convert( ba,
+            WarpX::GetInstance().getEfield_fp(0,1).ixType().toIntVect() ), dm, 2, nge );
+        pml_E_fp[2] = std::make_unique<MultiFab>(amrex::convert( ba,
+            WarpX::GetInstance().getEfield_fp(0,2).ixType().toIntVect() ), dm, 2, nge );
+    }
```

Comment:
Could we do the following to make the code simpler?
```suggestion
    {
        int const ncomp = (do_dive_cleaning) ? 3 : 2;
        pml_E_fp[0] = std::make_unique<MultiFab>(amrex::convert( ba,
            WarpX::GetInstance().getEfield_fp(0,0).ixType().toIntVect() ), dm, ncomp, nge );
        pml_E_fp[1] = std::make_unique<MultiFab>(amrex::convert( ba,
            WarpX::GetInstance().getEfield_fp(0,1).ixType().toIntVect() ), dm, ncomp, nge );
        pml_E_fp[2] = std::make_unique<MultiFab>(amrex::convert( ba,
            WarpX::GetInstance().getEfield_fp(0,2).ixType().toIntVect() ), dm, ncomp, nge );
    }
```